src/engine/order.spec.ts

```diff
@@ -409,14 +410,152 @@ describe('OrderEngine', () => {
+    mockPlaceOrder.mockReturnValueOnce(order).mockReturnValueOnce(false)
+    mockCheckOrder.mockReturnValueOnce(order).mockReturnValueOnce({ ...order, status: 'closed' })
+    mockGetOpenOrder
+      .mockReturnValueOnce(order)
+      .mockReturnValueOnce(order)
+      .mockReturnValueOnce(order)
+
+    const executeBuy = jest.spyOn(orderEngine, 'executeBuy')
+    const checkOrder = jest.spyOn<any, any>(orderEngine, 'checkOrder')
+    const cancelOrder = jest.spyOn<any, any>(orderEngine, 'cancelOrder').mockImplementationOnce(() => true)
+
+    await orderEngine.executeBuy()
+
+    expect(executeBuy).toHaveBeenCalledTimes(1)
+    expect(cancelOrder).toHaveBeenCalledTimes(0)
+    expect(checkOrder).toHaveBeenCalledTimes(2)
+  })
+
+  test('re-execute order if slippage (sell)', async () => {
+    mockAmountToPrecision.mockReturnValueOnce(mockAsset)
+    mockPriceToPrecision.mockReturnValueOnce(mockPrice).mockReturnValueOnce(mockPrice - 1)
+
+    const order = { id: mockId, status: 'open', side: 'sell', price: mockPrice }
+    mockPlaceOrder.mockReturnValueOnce(order).mockReturnValueOnce(false)
+    mockCheckOrder.mockReturnValueOnce(order)
+    mockGetOpenOrder
+      .mockReturnValueOnce(order)
+      .mockReturnValueOnce(order)
+      .mockReturnValueOnce(order)
+
+    const executeSell = jest.spyOn(orderEngine, 'executeSell')
+    const checkOrder = jest.spyOn<any, any>(orderEngine, 'checkOrder')
+    const cancelOrder = jest.spyOn<any, any>(orderEngine, 'cancelOrder').mockImplementationOnce(() => true)
+
+    await orderEngine.executeSell()
+
+    expect(executeSell).toHaveBeenCalledTimes(2)
+    expect(cancelOrder).toHaveBeenCalledTimes(1)
+    expect(cancelOrder).toHaveBeenCalledWith(mockId)
+    expect(checkOrder).toHaveBeenCalledTimes(1)
+  })
+
+  test('re-execute order if slippage (sell) w/ tolerance', async () => {
+    // @ts-ignore
+    orderEngine.orderSlippageAdjustmentTolerance = 1
+
+    mockAmountToPrecision.mockReturnValueOnce(mockAsset)
+    mockPriceToPrecision.mockReturnValueOnce(mockPrice).mockReturnValueOnce(mockPrice - 2)
+
+    const order = { id: mockId, status: 'open', side: 'sell', price: mockPrice }
+    mockPlaceOrder.mockReturnValueOnce(order).mockReturnValueOnce(false)
+    mockCheckOrder.mockReturnValueOnce(order)
+    mockGetOpenOrder
+      .mockReturnValueOnce(order)
+      .mockReturnValueOnce(order)
+      .mockReturnValueOnce(order)
+
+    const executeSell = jest.spyOn(orderEngine, 'executeSell')
+    const checkOrder = jest.spyOn<any, any>(orderEngine, 'checkOrder')
+    const cancelOrder = jest.spyOn<any, any>(orderEngine, 'cancelOrder').mockImplementationOnce(() => true)
+
+    await orderEngine.executeSell()
+
+    expect(executeSell).toHaveBeenCalledTimes(2)
+    expect(cancelOrder).toHaveBeenCalledTimes(1)
+    expect(cancelOrder).toHaveBeenCalledWith(mockId)
+    expect(checkOrder).toHaveBeenCalledTimes(1)
+  })
+
+  test("don't re-execute order if slippage (sell) w/ tolerance", async () => {
```

Comment:
Use `'` instead of `"`

Reply:
I think the lint rule is to prefer `"` over `'` if you need to escape an `'` in the string.